lib/models/device-type.ts

```diff
@@ -0,0 +1,214 @@
+import * as errors from 'balena-errors';
+
+const getDeviceTypeModel = function (deps: InjectedDependenciesParam) {
+	const { pine } = deps;
+
+	const exports = {
+		/**
+		 * @summary Get all deviceTypes
+		 * @name getAll
+		 * @public
+		 * @function
+		 * @memberof balena.models.deviceTypeType
+		 *
+		 * @param {Object} [options={}] - extra pine options to use
+		 * @fulfil {Object[]} - device types
+		 * @returns {Promise}
+		 *
+		 * @description
+		 * This method returns all supported device types.
+		 *
+		 * @example
+		 * balena.models.deviceTypeType.getAll().then(function(deviceTypes) {
+		 * 	console.log(deviceTypes);
+		 * });
+		 *
+		 * @example
+		 * balena.models.deviceType.getAll({ $select: ['name', 'slug'] }).then(function(deviceTypes) {
+		 * 	console.log(deviceTypes);
+		 * })
+		 *
+		 * @example
+		 * balena.models.deviceType.getAll(function(error, deviceTypes) {
+		 * 	if (error) throw error;
+		 * 	console.log(deviceTypes);
+		 * });
+		 */
+		async getAll(options?: PineOptions<DeviceType>): Promise<DeviceType[]> {
+			if (options == null) {
+				options = {};
+			}
+
+			const deviceType = await pine.get({
+				resource: 'device_type',
+				options: mergePineOptions({ $orderby: 'name asc' }, options),
+			});
+			return deviceType;
+		},
+
+		/**
+		 * @summary Get a deviceType by slug or name
+		 * @name getBySlugOrName
+		 * @public
+		 * @function
+		 * @memberof balena.models.deviceTypeType
+		 *
+		 * @param {String} slugOrName - deviceType slug
+		 * @fulfil {Object} - device manifest
+		 * @returns {Promise}
+		 *
+		 * @example
+		 * balena.models.deviceType.getBySlugOrName('raspberry-pi').then(function(manifest) {
+		 * 	console.log(manifest);
+		 * });
+		 *
+		 * @example
+		 * balena.models.deviceType.getBySlugOrName('raspberry-pi', function(error, manifest) {
+		 * 	if (error) throw error;
+		 * 	console.log(manifest);
+		 * });
+		 */
+		getBySlugOrName: async (slugOrName: string): Promise<DeviceType> => {
+			const deviceTypes = await exports.getAll();
+			const deviceType = deviceTypes.find(
+				(dt) => dt.name === slugOrName || dt.slug === slugOrName,
+			);
+			if (deviceType == null) {
+				throw new errors.BalenaInvalidDeviceType(slugOrName);
+			}
+			return deviceType;
+		},
+
+		/**
+		 * @summary Get display name for a device
+		 * @name getDisplayName
+		 * @public
+		 * @function
+		 * @memberof balena.models.deviceType
+		 *
+		 * @param {String} deviceTypeSlug - device type slug
+		 * @fulfil {String} - device display name
+		 * @returns {Promise}
+		 *
+		 * @example
+		 * balena.models.deviceType.getDisplayName('raspberry-pi').then(function(deviceTypeName) {
+		 * 	console.log(deviceTypeName);
+		 * 	// Raspberry Pi
+		 * });
+		 *
+		 * @example
+		 * balena.models.deviceType.getDisplayName('raspberry-pi', function(error, deviceTypeName) {
+		 * 	if (error) throw error;
+		 * 	console.log(deviceTypeName);
+		 * 	// Raspberry Pi
+		 * });
+		 */
+		getDisplayName: async (
```

Comment:
How about using `deviceType.getDisplayName` & `deviceType.getDeviceSlug` as the implementation for the respective `device.*` methods?